datafusion/physical-expr/src/udf.rs

```diff
@@ -28,8 +29,22 @@ use std::sync::Arc;
 pub fn create_physical_expr(
     fun: &ScalarUDF,
     input_phy_exprs: &[Arc<dyn PhysicalExpr>],
-    return_type: DataType,
+    input_schema: &Schema,
+    args: &[Expr],
+    input_dfschema: &DFSchema,
 ) -> Result<Arc<dyn PhysicalExpr>> {
+    let input_expr_types = input_phy_exprs
```

Comment:
I think we need to check the signature for udf too.

Reply:
I think the idea was at this point the type checking had already been done, so there is no reason to check the types again. 

However, I looked at how non udf functions work, and they also seem to check the data types again, so the change in this PR is consistent:

https://github.com/apache/arrow-datafusion/blob/6710e6db38d3f5292e6d9506c70c153c2e76f339/datafusion/physical-expr/src/functions.rs#L62-L84

Reply:
Do you mean they are checked in logical expr once, so we dont double check them in physical expr?
If so, we can remove both in physical expr

Reply:
I think we still need to check signature if someone create physcial expr directly (might be possible?).

Reply:
let me merge this first and move on #9504